### PR TITLE
all: fix misspelled comments

### PR DESCRIPTION
**Pull request by tsachiherman**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [X] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [X] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [X] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

This PR fix misspelling in the source code.

### Why

Misspelling in the source code aren't perceived as a production grade, professional code.

### Known limitations

N/A

### CODE REVIEW

File: clients/horizonclient/client.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func (c *Client) stream(
 						//
 						// In the former case, that (again) should never happen in Horizon, we need to
 						// check if there are any events we need to decode. We do this in the `if`
-						// statement below just in case if Horizon behaviour changes in a future.
+						// statement below just in case if Horizon behavior changes in a future.
```
Comment (paulbellamy) [inline clients/horizonclient/client.go:200]: Not a misspelling 😂

**Review by paulbellamy — APPROVED** [review]

File: clients/horizonclient/client.go (hunk 1 above)
Comment (tsachiherman) [reply clients/horizonclient/client.go:200]: I wasn't that creative myself.. I aligned these with the linter `misspell`

**Comment by 2opremio** [comment]
@tsachiherman now that misspellings are clean, would you consider adding a `mispell ` check to CI?

**Comment by tsachiherman** [comment]
> @tsachiherman now that misspellings are clean, would you consider adding a `mispell ` check to CI?

It's already enabled; however, it would add a comment to the PR only if you added a misspleed text in the PR.

if you want to run all the enabled linters against the codebase, you can run:
```sh
? golangci-lint run
```

**Pull request merged**